### PR TITLE
add baby_monster_group

### LINKED ISSUE #75830: Allow ``baby_monster`` to support monstergroups
### Is your feature request related to a problem? Please describe.

I have been endlessly complaining about why vertebrates should not upgrade into mutants. I stand by this notion, but it would be nice to allow the mutant population to increase in older maps anyway. This could be achieved by giving a low chance for vertebrates with mutant variants to give birth to a mutant version of themselves instead of a normal baby. 

This cannot be currently achieved, as ``baby_monster`` does not support monstergroups.

### Solution you would like.

Support for monstergroups in ``baby_monster``

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Close #75830 
#### Describe the solution
Add a new field, that picks monsters from mongroup and spawns them as babies
#### Testing
~~I can't! I can't find any way that confirms it works, nor any way that confirms that any another method of reproduction works! What the heck?~~ For whatever reason for monsters to actually reproduce you need to visit the place twice. Some troll logic, but it shows it works: cows with attached moose mongroup do spawn monsters from moose mongroup
image

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Speculation regarding reproduction "logic" (note: no actual facts!): Need to enter the reality bubble for the impregnation spores to get a chance to do their jobs (no such spores exist, but their "job" may well do) to set off a pregnancy timer, with a second reality bubble entry to effect the birth.
You need a first visit to get eggs laid, and a second one to get them to hatch.

It may also just be a load of hogwash...

**Comment by Brambor** [comment]
That sounds like a bug to be reported as a separate issue. To confirm: If I stay in a reality bubble with eggs, they never hatch?

**Comment by PatrikLundell** [comment]
No, I didn't mean to say you'd have to leave in between, just that you'd have to be in the bubble to "initiate" the process as well to finish it. Basically that neither egg laying nor hatching happens outside of the bubble, with the speculation that "impregnation" may similarly require bubble presence.

File: src/monmove.cpp [hunk 1]
```diff
@@ -415,7 +416,14 @@ void monster::anger_cub_threatened( monster_plan &mon_plan )
...
 
     for( monster &tmp : g->all_monsters() ) {
+        bool is_baby = false;
         if( type->baby_monster == tmp.type->id ) {
+            is_baby = true;
+        }
+        if( MonsterGroupManager::IsMonsterInGroup( type->baby_monster_group, tmp.type->id ) ) {
+            is_baby = true;
+        }
+        if( is_baby ) {
```
Comment (Brambor) [inline src/monmove.cpp:426]:
```suggestion
        const bool is_baby = type->baby_monster == tmp.type->id
                             || MonsterGroupManager::IsMonsterInGroup( type->baby_monster_group, tmp.type
        if( is_baby ) {
```

**Review by Brambor — COMMENTED** [review]
Simpler/Shorter bool logic.

File: src/monmove.cpp (hunk 1 above)
Comment (GuardianDll) [reply src/monmove.cpp:426]: should it be simplified further as 
```suggestion
        if( type->baby_monster == tmp.type->id || MonsterGroupManager::IsMonsterInGroup( type->baby_monster_group, tmp.type ) ) {
```
?

Comment (Brambor) [reply src/monmove.cpp]: It can be. It is up to the programmer. It is more readable sometimes, to have a named const bool.

**Pull request merged**